### PR TITLE
Feat(web): Add simple Modal component #DS-211

### PR DESCRIPTION
**Pull request by crishpeen**

### CODE REVIEW

**Comment by literat** [comment]
https://www.smashingmagazine.com/2022/04/cta-modal-build-web-component/ Just for inspiration and further reading.

File: packages/web/src/scss/components/Modal/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Modal
+
+Modal acts as dialog for any content.
+
+```html
+<button
```
Comment (literat) [inline packages/web/src/scss/components/Modal/README.md:6]: I am missing the backdrop in the example. Where should I place it?

**Review by literat — CHANGES_REQUESTED** [review]

File: packages/web/src/scss/components/Modal/README.md (hunk 1 above)
Comment (crishpeen) [reply packages/web/src/scss/components/Modal/README.md:6]: It's the same as in Header. So backdrop is added in JS.

**Review by literat — APPROVED** [review]

File: packages/web/src/scss/components/Modal/README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  data-target="#modal-example-1"
+  aria-controls="modal-example-1"
+  aria-expanded="false"
+>
+  Open Modal Example 1
+</button>
+<div id="modal-example-1" class="Modal">
+  <div class="Modal__dialog">
+    <div class="Modal__content">
+      <div class="Modal__header">
```
Comment (adamkudrna) [inline packages/web/src/scss/components/Modal/README.md:19]: I'm a little bit confused by the structure. This is what I'd expect:

```html
<div class="Modal">
  <div class="Modal__dialog">
    <div class="Modal__header"></div>
    <div class="Modal__content"></div>
  </div>
</div>
```

It becomes more obvious when I use `<dialog>` element:

```html
<div class="Modal">
  <dialog class="Modal__dialog">
    <div class="Modal__header"></div>
    <div class="Modal__content"></div> <!-- or Modal__body? -->
    <div class="Modal__footer"></div> <!-- future -->
  </dialog>
</div>
```

Could you please investigate if we really need the extra `<div>` around the actual dialog box?

File: packages/web/src/scss/components/Modal/_Modal.scss [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+@use '../../tools/accessibility';
+@use '../../tools/breakpoint';
+@use '../../tools/svg';
+@use 'theme';
+
+.Modal {
+    position: fixed;
+    top: 0;
+    overflow-x: hidden;
+    overflow-y: auto;
```
Comment (adamkudrna) [inline packages/web/src/scss/components/Modal/_Modal.scss:11]: To my surprise, it seems `-webkit-overflow-scrolling: touch` is not necessary any longer…

> Added support for one-finger accelerated scrolling to all frames and `overflow:scroll` elements eliminating the need to set `-webkit-overflow-scrolling: touch`.

https://developer.apple.com/documentation/safari-release-notes/safari-13-release-notes

File: packages/web/src/scss/components/Modal/_theme.scss [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+$content-border-radius: tokens.$radius-200;
+$content-background-color: tokens.$background-basic;
+$content-shadow: tokens.$shadow-300;
+
+$header-spacing: tokens.$space-600;
+
+$backdrop-background-color: tokens.$background-backdrop;
+
+$icon-size: tokens.$space-700;
+$icon-close: url('data:image/svg+xml,<svg width="24" height="24" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M18.3007 5.70997C17.9107 5.31997 17.2807 5.31997 16.8907 5.70997L12.0007 10.59L7.1107 5.69997C6.7207 5.30997 6.0907 5.30997 5.7007 5.69997C5.3107 6.08997 5.3107 6.71997 5.7007 7.10997L10.5907 12L5.7007 16.89C5.3107 17.28 5.3107 17.91 5.7007 18.3C6.0907 18.69 6.7207 18.69 7.1107 18.3L12.0007 13.41L16.8907 18.3C17.2807 18.69 17.9107 18.69 18.3007 18.3C18.6907 17.91 18.6907 17.28 18.3007 16.89L13.4107 12L18.3007 7.10997C18.6807 6.72997 18.6807 6.08997 18.3007 5.70997Z" fill="#{tokens.$action-primary-default}"/></svg>');
```
Comment (adamkudrna) [inline packages/web/src/scss/components/Modal/_theme.scss:23]:
```suggestion
$icon-close: url('data:image/svg+xml,<svg width="24" height="24" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M18.3007 5.70997C17.9107 5.31997 17.2807 5.31997 16.8907 5.70997L12.0007 10.59L7.1107 5.69997C6.7207 5.30997 6.0907 5.30997 5.7007 5.69997C5.3107 6.08997 5.3107 6.71997 5.7007 7.10997L10.5907 12L5.7007 16.89C5.3107 17.28 5.3107 17.91 5.7007 18.3C6.0907 18.69 6.7207 18.69 7.1107 18.3L12.0007 13.41L16.8907 18.3C17.2807 18.69 17.9107 18.69 18.3007 18.3C18.6907 17.91 18.6907 17.28 18.3007 16.89L13.4107 12L18.3007 7.10997C18.6807 6.72997 18.6807 6.08997 18.3007 5.70997Z" fill="#{tokens.$text-primary-default}"/></svg>');
```

image

**Review by adamkudrna — APPROVED** [review]
One last thing and it's ✅.

**Pull request merged**